tests/client-dbus/tests/udev/test_udev.py

```diff
@@ -266,7 +266,7 @@ def _simple_initial_discovery_test(self, *, key_description=None):
 
         settle()
 
-        with ServiceContextManager():
+        with OptionalKeyServiceContextManager(key_data) as key_description:
```

Comment:
We want both services to be ```OptionalKeyServiceContextManagers``` in this test, although it won't change the behavior for the second at this time.

Reply:
If you're looking to repopulate the key, we'll have to make changes to the `KernelKey` structure because it randomly generates a key description so the key description of the key available will not be the same as what's stored in the metadata.

Reply:
Whoops! Yes, that is what we really need to do.

Reply:
This may not be relevant for this PR but it may be in future tests so should we modify this now?

Reply:
@jbaublitz Please give that a try, we will definitely need it.